esmvalcore/preprocessor/_regrid.py

```diff
@@ -384,6 +384,18 @@ def extract_location(cube, location, scheme):
                          scheme)
 
 
+def _check_grid_discontiguities(cube, scheme):
+    """Check if there are grid discontiguities and set use_src_mask to True."""
+    try:
+        discontiguities = iris.util.find_discontiguities(cube)
+    except NotImplementedError:
+        pass
+    else:
+        if discontiguities.any():
+            scheme['use_src_mask'] = True
```

Comment:
Is it possible for users to provide the value of `use_src_mask` in the recipe? If yes, it would be good to at least log a message that we're changing their setting here.